### PR TITLE
add local mode

### PR DESCRIPTION
**Pull request by Oliver-Hanikel**
### Summary
This pull request adds the local_mode. When in local_mode no external CDNs get used as requested in #50.

### Details
I added local_mode to `config.py`. If `cfg.local_mode` is `True` the web dependencies get downloaded, if they aren't already on disk. For this I added `requests` to the `requirements.txt`. The templates now use the `system.web_deps` dictionary to get the correct web path for the dependencies.

### Checks
- [x] In case of new feature, add short overview in ```docs/<corresponding file>``` 
- [x] Tested changes

### CODE REVIEW

**Comment by Linbreux** [comment]
Hi @Oliver-Hanikel
Thanks for the PR. I'll check it this week ;)

File: config.py [hunk 1]
```diff
@@ -62,3 +64,5 @@ def __init__(self):
 
         self.protect_edit_by_password = os.getenv("PROTECT_EDIT_BY_PASSWORD") or yaml_config["protect_edit_by_password"] or PROTECT_EDIT_BY_PASSWORD
         self.password_in_sha_256 = os.getenv("PASSWORD_IN_SHA_256") or yaml_config["password_in_sha_256"] or PASSWORD_IN_SHA_256
+
+        self.local_mode = (os.getenv("LOCAL_MODE").lower() in ["true", "yes"]) or yaml_config["local_mode"] or LOCAL_MODE
```
Comment (Linbreux) [inline config.py:68]: When the env var LOCAL_MODE is not set the wiki does not start. This is because of the `lower()` function

**Review by Linbreux — CHANGES_REQUESTED** [review]
Hi this looks good! There is only a small bug when the env var is not set

File: config.py (hunk 1 above)
Comment (Oliver-Hanikel) [reply config.py:68]: I will fix it tomorrow

**Comment by Oliver-Hanikel** [comment]
Seems like requests dropped support for python 3.6
https://pypi.org/project/requests/

**Comment by Linbreux** [comment]
> Seems like requests dropped support for python 3.6
> https://pypi.org/project/requests/

Yes indeed, maybe `requests 2.27.1` would also work, this one supports 3.6 😀

**Review by Linbreux — APPROVED** [review]

**Comment by Linbreux** [comment]
Hi @Oliver-Hanikel 
Thanks for the work you have done! This is a great new feature!

**Pull request merged**